pkg/cluster/config/v1alpha2/conversion.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Code generated by conversion-gen. DO NOT EDIT.
+
+package v1alpha2
+
+import (
+	conversion "k8s.io/apimachinery/pkg/conversion"
+	config "sigs.k8s.io/kind/pkg/cluster/config"
+)
+
+func Convert_v1alpha2_Config_To_config_Cluster(in *Config, out *config.Cluster, s conversion.Scope) error {
+	for _, node := range in.Nodes {
+		// TODO(bentheelder): if v1alpha3 does re-gain per-node patches, drop
+		// this snippet
+		// Copy config patches from nodes to the cluster level, since we only
+		// generate a cluster-wide config anyhow
+		for _, patch := range node.KubeadmConfigPatches {
+			out.KubeadmConfigPatches = append(out.KubeadmConfigPatches, patch)
+		}
+		for _, patch := range node.KubeadmConfigPatchesJSON6902 {
+			out.KubeadmConfigPatchesJSON6902 = append(out.KubeadmConfigPatchesJSON6902, patch)
+		}
+
+		// skip now-implicit external load balancers
+		if node.Role == ExternalLoadBalancerRole {
+			continue
+		}
+
+		// always convert the node
+		convertedNode := config.Node{}
+		if err := Convert_v1alpha2_Node_To_config_Node(&node, &convertedNode, s); err != nil {
+			return err
+		}
+		out.Nodes = append(out.Nodes, convertedNode)
+
+		// handle additional replicas if any
+		if node.Replicas != nil && *node.Replicas > 1 {
+			for i := int32(1); i < *node.Replicas; i++ {
+				out.Nodes = append(out.Nodes, *convertedNode.DeepCopy())
+			}
+		}
+	}
+	return nil
+}
+
+func Convert_v1alpha2_Node_To_config_Node(in *Node, out *config.Node, s conversion.Scope) error {
+	return autoConvert_v1alpha2_Node_To_config_Node(in, out, s)
+}
+
+func Convert_config_Cluster_To_v1alpha2_Config(in *config.Cluster, out *Config, s conversion.Scope) error {
+	// TODO(bentheelder): try to convert kubeadm config patches?
+	// Not sure if that makes sense since these aren't cluster wide, and we don't
+	// actually roundtrip anywho? :thinking:
```

Comment:
seems OK to me to exclude the patches..
proper rountrips would be nice to have in future versions though.

Reply:
we could include them currently by shoe-horning them onto one of the nodes, but I'd rather revisit it later given we don't depend on this and that might not be the right answer 🤔

Reply:
I have a WIP to generate a config file per node, for better version skew. I'll revisit per-node patches as an _actually_ supported thing in addition to global patches as part of that.